### PR TITLE
Added Hard Times Keys Upgrade

### PR DESCRIPTION
**Pull request by BowlsforRick**

### CODE REVIEW

**Comment by BowlsforRick** [comment]
Trainers have now been added

**Comment by LinosM** [comment]
* Pull back the chord sustains on Expert/Hard 5L

* Medium can match Hard reductions during the intro and all repeated sections

* Consider animating the left hand playing the bass synths from the backing track during the chorus

* 51-66: When Hard is playing a constant stream of 8th notes, consider removing the last note of each measure. So remove any note on X.4.50 during this part

* 76-92: The reductions here for Hard/Medium are inconsistent between Pro/5L
  * Hard: 
  76-81 (chorus_3b): Keep every note from expert except for 76.4.50, 78.4.50, 80.4.50
  84-91 (outro): On 5L, move the notes on XX.3.50 1/16th over to XX.3.75. Afterwards have Pro match this reduction

  * Medium:
  76-81 (chorus_3b): Have Pro match 5L's reductions
  84-91 (outro): Restore the notes on XX.3.75, restore the note on 87.4.00, trim this stubby sustain on Pro 87.3.00 into a single note

* 41.3.00: Restore this note on Easy Pro

* Try to keep trainers no longer than 4 measures

   * Have trainer 2 go from 20.1.00 to 24.1.00
   * Have trainer 4: 76.1.00 to 80.1.00
   * Create a new trainer from 80.1.00 to 84.1.00

**Comment by BowlsforRick** [comment]
Updated Midi file with the recommended changes above. I've added LH animations and I've fixed up the inconsistencies between 5L/Pro for easy as well.

**Comment by LinosM** [comment]
Outro Hard is still not matched...

Hard 5L: Move these notes to the right a 1/16th
84.4.25, 85.4.25, 86.4.25, 88.4.25, 89.4.25, 90.4.25

Hard Pro: Restore these notes
84.1.50, 84.2.50, 85.1.50, 85.2.50, 86.1.50, 86.2.50, 87.1.50, 87.2.50
88.1.50, 88.2.50, 89.1.50, 89.2.50, 90.1.50, 90.2.50, 91.1.50, 91.2.50, 91.3.50

**Comment by BowlsforRick** [comment]
Fixed hard reductions on 5L and pro

File: Pro Keys (No New Audio)/Rock Band 4 DLC/hardtimes/songs.dta [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   (tuning_offset_cents 0.0)
+   (game_origin 'rb4_dlc')
+   (album_track_number 1)
+   (album_name "After Laughter")
+   (decade 'the10s')
+   (downloaded TRUE)
+   (base_points 0)
+   (vocal_tonic_note 5)
+   (keys_author "JIManji, Ultimate_MANG0, BowlsforRick")
+   (song_tonality 1) - Only set this for pro guitar upgrades and if you know what you are doing
```
Comment (LinosM) [inline Pro Keys (No New Audio)/Rock Band 4 DLC/hardtimes/songs.dta:61]:
```suggestion
   (song_tonality 1) ;- Only set this for pro guitar upgrades and if you know what you are doing
```
Make sure to add a semicolon here so this comment doesn't get read by the game

**Pull request merged**